api/src/main/java/org/apache/iceberg/RewriteFiles.java

```diff
@@ -42,4 +42,13 @@
    * @return this for method chaining
    */
   RewriteFiles rewriteFiles(Set<DataFile> filesToDelete, Set<DataFile> filesToAdd);
+
+  /**
+   * Add a rewrite that replaces one set of deletes with another that contains the same deleted rows.
+   *
+   * @param deletesToDelete files that will be replaced, cannot be null or empty.
+   * @param deletesToAdd files that will be added, cannot be null or empty.
+   * @return this for method chaining
+   */
+  RewriteFiles rewriteDeletes(Set<DeleteFile> deletesToDelete, Set<DeleteFile> deletesToAdd);
```

Comment:
Before we get this PR merged,  we must merge this PR first (https://github.com/apache/iceberg/pull/2294).  Because PR 2294 has extended the API `RewriteFiles` to accept both data files & delete files.  This PR is trying to convert equality-deletions to pos-deletions , which is actually a special case of the PR 2294.  After we merged PR 2294,  we could rebase this PR. 

For getting https://github.com/apache/iceberg/pull/2294 merged,  I need a committer to review it. FYI @rdblue , @yyanyy .